src/main/java/org/mule/module/apikit/validation/attributes/HeadersValidator.java

```diff
@@ -53,31 +61,54 @@ public HeadersValidator(Action action) {
     if (headersStrictValidation) {
       validateHeadersStrictly(incomingHeaders);
     }
-    MultiMap<String, String> copyIncomingHeaders = incomingHeaders;
+    MultiMap<String, String> copyIncomingHeaders = emptyMultiMap();
```

Comment:
Why not copy here all the incoming ? instead of emptyMultiMap

Reply:
For the sake of performance I tried to avoid copying the whole map if no necessary. It is not needed if the API doesn't have an array header or default values.
I added `getMutableCopy` in order to avoid copying the whole map for every header in the action as well.